erpnext/stock/stock_ledger.py

```diff
@@ -1401,9 +1405,14 @@ def get_previous_sle_of_current_voucher(args, operator="<", exclude_current_vouc
 
 	args["time_format"] = "%H:%i:%s"
 	if not args.get("posting_date"):
-		args["posting_date"] = "1900-01-01"
-	if not args.get("posting_time"):
-		args["posting_time"] = "00:00"
+		args["posting_datetime"] = "1900-01-01 00:00:00"
+
+	if not args.get("posting_datetime"):
+		args["posting_datetime"] = get_combine_datetime(args["posting_date"], args["posting_time"])
+
+	if operator == "<=":
+		# Add 1 second to handle millisecond for less than and equal to condition
+		args["posting_datetime"] = add_to_date(args["posting_datetime"], seconds=1)
```

Comment:
This will not work as `args` is a dictionary you'll end up overriding the previous conditions.

Reply:
Yes it's working because of order by and limit, but maybe need to think of ms case.